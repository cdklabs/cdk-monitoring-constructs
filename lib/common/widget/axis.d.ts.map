{"version":3,"file":"axis.d.ts","sourceRoot":"","sources":["axis.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,UAAU,EAAE,MAAM,4BAA4B,CAAC;AAExD;;GAEG;AACH,eAAO,MAAM,sBAAsB,EAAE,UAIpC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,+BAA+B,EAAE,UAK7C,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,sBAAsB,EAAE,UAIpC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,uBAAuB,EAAE,UAIrC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,uBAAuB,EAAE,UAIrC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,iBAAiB,EAAE,UAI/B,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,gBAAgB,EAAE,UAI9B,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,wBAAwB,EAAE,UAKtC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,gBAAgB,EAAE,UAI9B,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,qBAAqB,EAAE,UAKnC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,qBAAqB,EAAE,UAInC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,yBAAyB,EAAE,UAIvC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,yBAAyB,EAAE,UAIvC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,yBAAyB,EAAE,UAIvC,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,+BAA+B,EAAE,UAI7C,CAAC;AAEF;;GAEG;AACH,eAAO,MAAM,uBAAuB,EAAE,UAIrC,CAAC"}