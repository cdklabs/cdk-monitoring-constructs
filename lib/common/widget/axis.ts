import { YAxisProps } from "aws-cdk-lib/aws-cloudwatch";

/**
 * Y-Axis showing percentage, ranged from 0.
 */
export const PercentageAxisFromZero: YAxisProps = {
  min: 0,
  label: "%",
  showUnits: false,
};

/**
 * Y-Axis showing percentage, ranged from 0 to 100.
 */
export const PercentageAxisFromZeroToHundred: YAxisProps = {
  min: 0,
  max: 100,
  label: "%",
  showUnits: false,
};

/**
 * Y-Axis showing percentage, ranged up to 100.
 */
export const PercentageAxisUpToHundred: YAxisProps = {
  max: 100,
  label: "%",
  showUnits: false,
};

/**
 * Y-Axis showing time in milliseconds, ranged from 0.
 */
export const TimeAxisMillisFromZero: YAxisProps = {
  min: 0,
  label: "ms",
  showUnits: false,
};

/**
 * Y-Axis showing time in seconds, ranged from 0.
 */
export const TimeAxisSecondsFromZero: YAxisProps = {
  min: 0,
  label: "sec",
  showUnits: false,
};

/**
 * Y-Axis showing time in minutes, ranged from 0.
 */
export const TimeAxisMinutesFromZero: YAxisProps = {
  min: 0,
  label: "min",
  showUnits: false,
};

/**
 * Y-Axis showing count, ranged from 0.
 */
export const CountAxisFromZero: YAxisProps = {
  min: 0,
  label: "Count",
  showUnits: false,
};

/**
 * Y-Axis showing count, ranged from 1.
 */
export const CountAxisFromOne: YAxisProps = {
  min: 1,
  label: "Count",
  showUnits: false,
};

/**
 * Y-Axis showing boolean, ranged from 0 to 1, where 1 means TRUE.
 */
export const BooleanAxisFromZeroToOne: YAxisProps = {
  min: 0,
  max: 1,
  label: "1 = True",
  showUnits: false,
};

/**
 * Y-Axis showing rate (relative count), ranged from 0.
 */
export const RateAxisFromZero: YAxisProps = {
  min: 0,
  label: "Rate",
  showUnits: false,
};

/**
 * Y-Axis showing rate (relative count), ranged from 0 to 1.
 */
export const RateAxisFromZeroToOne: YAxisProps = {
  min: 0,
  max: 1,
  label: "Rate",
  showUnits: false,
};

/**
 * Y-Axis showing size in bytes, ranged from 0.
 */
export const SizeAxisBytesFromZero: YAxisProps = {
  min: 0,
  label: "bytes",
  showUnits: false,
};

/**
 * Y-Axis showing size in kilobytes, ranged from 0.
 */
export const SizeAxisKiloBytesFromZero: YAxisProps = {
  min: 0,
  label: "kB",
  showUnits: false,
};

/**
 * Y-Axis showing size in megabytes, ranged from 0.
 */
export const SizeAxisMegaBytesFromZero: YAxisProps = {
  min: 0,
  label: "MB",
  showUnits: false,
};

/**
 * Y-Axis showing size in gigabytes, ranged from 0.
 */
export const SizeAxisGigaBytesFromZero: YAxisProps = {
  min: 0,
  label: "GB",
  showUnits: false,
};

/**
 * Y-Axis showing cost in Megabyte Millisecond, ranged from 0.
 */
export const MegabyteMillisecondAxisFromZero: YAxisProps = {
  min: 0,
  label: "MB*ms",
  showUnits: false,
};

/**
 * Y-Axis showing cost in USD, ranged from 0.
 */
export const CurrencyAxisUsdFromZero: YAxisProps = {
  min: 0,
  label: "USD",
  showUnits: false,
};
