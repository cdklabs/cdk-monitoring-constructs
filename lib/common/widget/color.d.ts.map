{"version":3,"file":"color.d.ts","sourceRoot":"","sources":["color.ts"],"names":[],"mappings":"AAAA;;GAEG;AACH,eAAO,MAAM,kBAAkB,YAAY,CAAC;AAC5C;;GAEG;AACH,eAAO,MAAM,oBAAoB,YAAY,CAAC;AAC9C;;GAEG;AACH,eAAO,MAAM,YAAY,YAAY,CAAC;AACtC;;GAEG;AACH,eAAO,MAAM,UAAU,YAAY,CAAC;AACpC;;GAEG;AACH,eAAO,MAAM,YAAY,YAAY,CAAC"}