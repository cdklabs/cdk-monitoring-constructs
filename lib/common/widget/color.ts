/**
 * color to indicate healthy state
 */
export const HealthyMetricColor = "#2ca02c";
/**
 * color to indicate unhealthy state
 */
export const UnhealthyMetricColor = "#d62728";
/**
 * color to indicate a warning
 */
export const WarningColor = "#ff9900";
/**
 * color to indicate an error
 */
export const ErrorColor = "#d13212";
/**
 * color to indicate neutral information
 */
export const NeutralColor = "#999999";
/**
 * color to indicate throttled state
 */
export const ThrottledColor = "#5928ed";
/**
 * color to indicate a started state (e.g. Step Function executions)
 */
export const StartedColor = "#0073e6";
/**
 * color to indicate a timed out state
 */
export const TimedOutColor = "#333333";
