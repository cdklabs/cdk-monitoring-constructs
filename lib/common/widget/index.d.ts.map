{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,QAAQ,CAAC;AACvB,cAAc,SAAS,CAAC;AACxB,cAAc,QAAQ,CAAC;AACvB,cAAc,SAAS,CAAC"}