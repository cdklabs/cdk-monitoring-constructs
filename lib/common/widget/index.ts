export * from "./axis";
export * from "./color";
export * from "./size";
export * from "./types";
