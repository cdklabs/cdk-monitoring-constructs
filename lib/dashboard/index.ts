export * from "./BitmapDashboard";
export * from "./DashboardRenderingPreference";
export * from "./DynamicDashboardSegment";
export * from "./DashboardSegment";
export * from "./DashboardWithBitmapCopy";
export * from "./DefaultDashboardFactory";
export * from "./DynamicDashboardFactory";
export * from "./DefaultWidgetFactory";
export * from "./IDashboardFactory";
export * from "./IDynamicDashboardFactory";
export * from "./IWidgetFactory";
export * from "./MonitoringNamingStrategy";
export * from "./SingleWidgetDashboardSegment";
export * from "./widget";
