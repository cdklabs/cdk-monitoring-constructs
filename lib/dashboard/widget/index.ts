export * from "./AlarmMatrixWidget";
export * from "./BitmapWidget";
export * from "./CustomWidget";
export * from "./HeaderWidget";
export * from "./KeyValueTableWidget";
export * from "./MonitoringHeaderWidget";
export * from "./StrictGraphWidget";
export * from "./UnofficialWidgets";
