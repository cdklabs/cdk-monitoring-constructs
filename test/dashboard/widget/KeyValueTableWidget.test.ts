import { KeyValueTableWidget } from "../../../lib/dashboard/widget";

test("snapshot test", () => {
  const data: [string, string][] = [
    ["name", "John Wick"],
    ["has", "a dog"],
  ];
  const widget = new KeyValueTableWidget(data);

  expect(widget.toJson()).toMatchSnapshot();
});
