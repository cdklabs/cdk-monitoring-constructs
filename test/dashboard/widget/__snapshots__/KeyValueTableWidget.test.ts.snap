// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Array [
  Object {
    "height": 3,
    "properties": Object {
      "background": undefined,
      "markdown": "| name| has
|---|---
| John Wick| a dog",
    },
    "type": "text",
    "width": 24,
    "x": undefined,
    "y": undefined,
  },
]
`;
